src/Microsoft.HttpRepl.Telemetry/TelemetryCommonProperties.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+
+namespace Microsoft.HttpRepl.Telemetry
+{
+    public class TelemetryCommonProperties
+    {
+        private readonly string _productVersion;
+
+        public TelemetryCommonProperties(
+            string productVersion,
+            Func<string, string> hasher = null,
+            Func<string> getMACAddress = null,
+            IDockerContainerDetector dockerContainerDetector = null,
+            IUserLevelCacheWriter userLevelCacheWriter = null)
+        {
+            _productVersion = productVersion;
+            _hasher = hasher ?? Sha256Hasher.Hash;
```

Comment:
For all the non-telemetry pieces (plus the ITelemetry/Telemetry object itself) we do have the ComposeDependencies in Program.cs that organizes them normally. I didn't think it was necessary to pull all of these up into that, especially given the hope that this would eventually be consolidated into a separate shared assembly which may or may not have that same pattern.